modules/imagetools/src/main/ts/ephox/imagetools/meta/ExifReader.ts

```diff
@@ -444,14 +452,23 @@ export class ExifReader {
       // Set binary reader pointer to beginning of the next tag
       let offset = IFD_offset + 2 + i * 12;
 
-      const tag = tags2extract[self.SHORT(offset)!];
+      const tagId = self.SHORT(offset);
+      if (tagId === null) {
+        throw new Error('Invalid Exif data.');
```

Comment:
I assume this is fine but I don't think we normally throw errors so looks very odd... As long as this gets picked up somewhere and doesn't break things?

Reply:
Every line I changed was due to the code ignoring errors. Most of them would have resulted in a typeerror being thrown. Some would have resulted in null being cast to zero. 

None of the errors should have been ignored at all...

Reply:
In the above case `null` would have caused tag to be `undefined` which would have caused the item to be skipped - it is still an error though just one we ignored.

Reply:
Another alternative would be to change the API to return Result which would be the functional way of dealing with all these errors.

Reply:
since this was already throwing an error - and a bad one - imo this is fine, we can log a ticket to consider making it a functional API?

Reply:
A yak was shaved, it's now functional.